test/distributed/fsdp/test_fsdp_optim_state.py

```diff
@@ -1668,6 +1668,68 @@ def forward(self, x):
             state_dicts[0], state_dicts[1], check_same_param_keys=True
         )
 
+    @skip_if_lt_x_gpu(2)
+    def test_optim_state_without_param_groups(self):
+        class SimpleModel(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                torch.manual_seed(0)
+                self.net1 = nn.Sequential(nn.Linear(2, 4), nn.ReLU())
+
+            def forward(self, x):
+                return self.net1(x)
+
+        model = FSDP(SimpleModel().cuda())
+        optim = torch.optim.Adam(model.parameters(), lr=1e-2)
+
+        # Train one step to save original optimizer state dict and original optimizer param groups.
+        batch = torch.rand(3, 2, device=torch.device("cuda"))
+        for param in model.parameters():
+            if param.requires_grad:
+                t = torch.zeros_like(param)
+                param.grad = torch.autograd.Variable(t)
+        optim.step()
+        loss = model(batch).sum()
+        loss.backward()
+
+        original_osd = deepcopy(optim.state_dict())
+        original_osd_no_param_groups = deepcopy(original_osd)
+        # manually remove param_groups from optimizer state dict
+        original_param_groups = deepcopy(
+            original_osd_no_param_groups.pop("param_groups")
+        )
+        # passing the osd without param_groups to FSDP
+        original_fsdp_state_dict = deepcopy(
+            FSDP.optim_state_dict(
+                model, optim, optim_state_dict=original_osd_no_param_groups
```

Comment:
when there were no param groups, would there be a crash previously? If that is the case, do we know why setting TorchRec `opt.save_param_groups(False)` did not result in a crash? cc @fegin

Reply:
There was a crash. Isn't that the reason we set `optim.save_param_groups` to True?

Reply:
Ah ok, I forgot whether we had it as true or false.